abm/projects/cooperative_signaling/cs_agent/cs_agent.py

```diff
@@ -165,172 +242,13 @@ def draw_update(self):
-                    self.vis_field_source_data[i]["proj_end"] = int(
-                        phi_target + proj_size / 2)
-                    self.vis_field_source_data[i]["proj_start_ex"] = int(
-                        phi_target - proj_size / 2)
-                    self.vis_field_source_data[i]["proj_end_ex"] = int(
-                        phi_target + proj_size / 2)
-                    self.vis_field_source_data[i]["proj_size_ex"] = proj_size
-                    if non_expl_agents is not None:
-                        if i < len_social:
-                            self.vis_field_source_data[i][
-                                "is_social_cue"] = True
-                        else:
-                            self.vis_field_source_data[i][
-                                "is_social_cue"] = False
-                    else:
-                        self.vis_field_source_data[i]["is_social_cue"] = True
-        # calculating visual exclusion if requested
-        if self.visual_exclusion:
-            self.exlude_V_source_data()
-        if non_expl_agents is not None:
-            # removing non-social cues from the source data after calculating
-            # exclusions
-            self.remove_nonsocial_V_source_data()
-        # sorting VPF source data according to visual angle
-        self.rank_V_source_data("vis_angle")
-        for k, v in self.vis_field_source_data.items():
-            vis_angle = v["vis_angle"]
-            phi_target = v["phi_target"]
-            proj_size = v["proj_size"]
-            proj_start = v["proj_start_ex"]
-            proj_end = v["proj_end_ex"]
-            # circular boundaries to the VPF as there is 360 degree vision
-            if proj_start < 0:
-                v_field[self.v_field_res + proj_start:self.v_field_res] = 1
-                proj_start = 0
-            if proj_end >= self.v_field_res:
-                v_field[0:proj_end - self.v_field_res] = 1
-                proj_end = self.v_field_res - 1
-            # weighing projection amplitude with rank information if requested
-            if not keep_distance_info:
-                v_field[proj_start:proj_end] = 1
-            else:
-                v_field[proj_start:proj_end] = v["meter"]
-
-        # post_processing and limiting FOV
-        v_field_post = np.flip(v_field)
-        v_field_post[phis < fov[0]] = 0
-        v_field_post[phis > fov[1]] = 0
-        return v_field_post
-
     def prove_velocity(self, velocity_limit=1):
         """Restricting the absolute velocity of the agent"""
-        vel_sign = np.sign(self.velocity)
-        if vel_sign == 0:
-            vel_sign = +1
-        if self.get_mode() == 'explore':
+        if self.agent_state == 'exploration':
             if np.abs(self.velocity) > velocity_limit:
                 # stopping agent if too fast during exploration
-                self.velocity = 1
+                # QUESTION: can we return velocity_limit instead of 1?
```

Comment:
@mezdahun I am not sure if this matters at all

Reply:
This is something from the static foraging because we wanted to keep the agent velocity constant all the time. In case we control this already in the corresponding parts of the cooperative signalling code this is not needed and can be simply deleted